### PR TITLE
feat: error out when config file contains 1.x config values

### LINKED ISSUE #22863: Detect and error out when users try passing 1.x config to the 2.x `influxd`
Users new to InfluxDB sometimes try to follow a 1.x tutorial using the 2.x binaries. A common sign this is happening is when a user tries to pass a 1.x `influxdb.conf` file to the binary instead of one of our supported formats for 2.x. We should detect when this happens and return a clear error message that guides the user towards 2.x configuration docs.

### PR DESCRIPTION
**Pull request by wbaker85**
Closes #22863

Throws an error on server startup if there are values in the config file ~which do not correspond to a valid configuration option~ which are determined to be from a 1.x config. Also adds debug logging if a config file is loaded.

### CODE REVIEW

**Comment by wbaker85** [comment]
Example when starting the server with a bad config value - `bind-address` used instead of `http-bind-address`:

```
./bin/darwin/influxd
error: unknown flags from config file at /Users/wbaker/influxdb/cmd/influxd/config.toml (see https://docs.influxdata.com/influxdb/latest/reference/config-options/ for supported flags): bind-address
```

If a full 1.x config is used, there will be many invalid options:

```
./bin/darwin/influxd
error: unknown flags from config file at /Users/wbaker/influxdb/cmd/influxd/config.toml (see https://docs.influxdata.com/influxdb/latest/reference/config-options/ for supported flags): coordinator.max-select-buckets,logging.level,http.flux-log-enabled,http.pprof-auth-enabled,http.ping-auth-enabled,http.pprof-enabled,coordinator.write-timeout,data.compact-throughput,http.bind-socket,http.auth-enabled,subscriber.write-concurrency,data.trace-logging-enabled,data.compact-throughput-burst,http.max-row-limit,coordinator.max-select-series,logging.suppress-logo,subscriber.http-timeout,tls.max-version,monitor.store-database,data.tsm-use-madv-willneed,udp,opentsdb,http.realm,http.bind-address,monitor.store-enabled,data.series-id-set-cache-size,data.wal-dir,data.cache-max-memory-size,data.cache-snapshot-memory-size,http.prom-read-auth-enabled,http.max-connection-limit,coordinator.log-queries-after,coordinator.max-concurrent-queries,monitor.store-interval,data.max-concurrent-compactions,data.series-file-max-concurrent-snapshot-compactions,continuous_queries.enabled,http.flux-enabled,data.dir,data.index-version,data.max-index-log-file-size,continuous_queries.run-interval,http.https-private-key,http.write-tracing,data.max-series-per-database,data.strict-error-handling,data.max-values-per-tag,data.cache-snapshot-write-cold-duration,meta.logging-enabled,http.unix-socket-enabled,http.https-enabled,bind-address,coordinator.max-select-point,logging.format,subscriber.enabled,data.validate-keys,shard-precreation.enabled,shard-precreation.check-interval,retention.enabled,meta.retention-autocreate,graphite,http.log-enabled,http.max-enqueued-write-limit,coordinator.query-timeout,http.enqueued-write-timeout,tls.min-version,subscriber.write-buffer-size,collectd,http.unix-socket-permissions,http.max-body-size,subscriber.ca-certs,retention.check-interval,subscriber.insecure-skip-verify,continuous_queries.log-enabled,http.debug-pprof-enabled,http.max-concurrent-write-limit,http.access-log-path,http.enabled,shard-precreation.advance-period,data.wal-fsync-delay,data.compact-full-write-cold-duration,continuous_queries.query-stats-enabled,meta.dir,http.suppress-write-log,http.shared-secret,http.https-certificate,data.query-log-enabled
```

**Review by lesam — CHANGES_REQUESTED** [review]
We should think some more about how this affects upgrades and rollbacks where the config changes.

**Comment by wbaker85** [comment]
Updated to check specifically for 1.x config values (and only 1.x config values).

File: cmd/influxd/launcher/cmd_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			want:   []string{"meta.something"},
+		},
+		{
+			name:   "invalid global flag",
+			config: "bind-address: value",
+			want:   []string{"bind-address"},
+		},
+		{
+			name:   "invalid global and dot",
+			config: "bind-address: value\nmeta.something: value",
```
[the remark this answers is not in the document]
Comment (wbaker85) [reply cmd/influxd/launcher/cmd_test.go]: Yeah, probably makes more sense to use those more directly now that this is specifically looking for 1x config values. Updated the tests to be more along those lines.

**Review by lesam — APPROVED** [review]
Thanks!

**Pull request merged**